### ISSUE TITLE
Add callbacks to monitor training progress and prevent overfitting

### ISSUE THREAD

**Issue by tallamjr** [tallamjr/astronet#47]
Currently, hyperparameter optimisation and training take place for a given number of epochs, it would be desirable to be able to implement an early stopping criteria if it is seem to be overfitting, or that the progress has plateaued.

`tensorflow.keras.callbacks` seem to be a good fit for this.

See [Coursera videos](https://www.coursera.org/learn/custom-models-layers-loss-functions-with-tensorflow/lecture/Qk2yU/built-in-callbacks) for details on how to implement
